### PR TITLE
`azurerm_dns_zone`: Remove non-existent `fqdn` attribute from docs

### LINKED ISSUE #10759: azurerm_dns_zone does not expose attribute fqdn
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

Terraform v0.14.7
+ provider registry.terraform.io/hashicorp/azurerm v2.49.0
+ provider registry.terraform.io/hashicorp/kubernetes v2.0.2
+ provider registry.terraform.io/hashicorp/random v3.1.0

### Affected Resource(s)

* `azurerm_dns_zone`

### Terraform Configuration Files

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://keybase.io/hashicorp

provider "azurerm" {
  features {

  }
}

resource "azurerm_resource_group" "rg" {
  name     = "my-resourcegroup"
  location = "west europe"
}

resource "azurerm_dns_zone" "environment" {
  name                = "subdomain.domain.com"
  resource_group_name = azurerm_resource_group.rg.name
}

output "cname_fqdn" {
  value = azurerm_dns_zone.environment.fqdn
}
```

### Debug Output

https://gist.github.com/michaelelleby/c717c8e8cabf3528c1e5dc59bcfd2240

### Expected Behaviour

Output should have included fqdn of dns zone, as documented in https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/dns_zone#attributes-reference

### Actual Behaviour

Error: Unsupported attribute

  on main.tf line 18, in output "cname_fqdn":
  18:   value = azurerm_dns_zone.environment.fqdn

### Steps to Reproduce

1. `terraform validate`

### References

* #0000

### PR DESCRIPTION
**Pull request by favoretti**
Fixes #10759

### CODE REVIEW

File: azurerm/internal/services/dns/dns_zone_resource.go [hunk 1]
```diff
@@ -253,6 +259,12 @@ func resourceDnsZoneRead(d *schema.ResourceData, meta interface{}) error {
 		return fmt.Errorf("setting `soa_record`: %+v", err)
 	}
 
+	fqdn := ""
+	if rsResp.Fqdn != nil {
+		fqdn = *rsResp.Fqdn
+	}
+	d.Set("fqdn", fqdn)
```
Comment (mbfrahry) [inline azurerm/internal/services/dns/dns_zone_resource.go:266]: Thoughts on shortening lines 262-266 to just. d.Set does a pretty good job of handling nil base types as long as we don't point to it. Lists and Sets are the ones we have to make sure we always set to at least an empty value

```suggestion
	d.Set("fqdn", rsResp.Fqdn)
```

**Review by mbfrahry — CHANGES_REQUESTED** [review]
Thanks for another one @favoretti, this looks good as well but we should add documentation too

**Comment by favoretti** [comment]
@mbfrahry As @manicminer pointed out - this actually makes no sense, since `name` property is exactly that.
I'll remove this and remove copy-paste error from the docs instead.

**Review by mbfrahry — APPROVED** [review]
LGTM!

**Pull request merged**